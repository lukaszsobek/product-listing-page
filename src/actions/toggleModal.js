import { TOGGLE_MODAL } from "../constants";

const toggleModal = modal => ({
    type: TOGGLE_MODAL,
    modal 
});

export default toggleModal;