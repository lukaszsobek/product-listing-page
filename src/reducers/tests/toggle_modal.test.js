import toggleModal from '../toggle_modal'

describe('toggleModal reducer', () => {
  const state = {
    modalState: {
      colorModalVisible: false,
      categoryModalVisible: false
    }
  }

  it('opens a modal if closed', () => {
    const action = {
      modal: 'ColorModal'
    }
    expect(toggleModal(state, action)).toMatchSnapshot()
  })

  it('closes a modal if already open', () => {
    state.modalState.colorModalVisible = true
    const action = {
      modal: 'ColorModal'
    }
    expect(toggleModal(state, action)).toMatchSnapshot()
  })

  it('closes other modal when this opened', () => {
    state.modalState.colorModalVisible = true
    const action = {
      modal: 'CategoryModal'
    }
    expect(toggleModal(state, action)).toMatchSnapshot()
  })
})
