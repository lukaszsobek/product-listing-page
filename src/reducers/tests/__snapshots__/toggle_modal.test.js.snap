// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleModal reducer closes a modal if already open 1`] = `
Object {
  "modalState": Object {
    "categoryModalVisible": false,
    "colorModalVisible": false,
  },
}
`;

exports[`toggleModal reducer closes other modal when this opened 1`] = `
Object {
  "modalState": Object {
    "categoryModalVisible": true,
    "colorModalVisible": false,
  },
}
`;

exports[`toggleModal reducer opens a modal if closed 1`] = `
Object {
  "modalState": Object {
    "categoryModalVisible": false,
    "colorModalVisible": true,
  },
}
`;
